spot_rl_experiments/spot_rl/envs/place_env.py

```diff
@@ -2,75 +2,243 @@
+                - time_taken (float): Time taken to place the object
+                - success (bool): Whether the place was successful
+                - place_target (np.array([x,y,z])): Place target in base frame
+                - ee_pos (np.array([x,y,z])): End effector position in base frame
+        """
+        success_list = []
+        for place_target in place_target_list:
+            start_time = time.time()
+            if self.use_policies:
+                observations = self.place_env.reset(place_target, is_local)
+                done = False
+
+                while not done:
+                    action = self.policy.act(observations)
+                    observations, _, done, _ = self.place_env.step(arm_action=action)
+
+            else:
+                # Get reset arm position (Is there a better way to do this without using place_env?????????????)
+                self.place_env.reset(place_target, is_local)
+
+                # End effector positions in base frame (as needed by the API)
+                curr_ee_pos = self.place_env.get_gripper_position_in_base_frame_spot()
+                goal_ee_pos = self.place_env.get_base_frame_place_target_spot()
+                intr_ee_pos = generate_intermediate_point(curr_ee_pos, goal_ee_pos)
+
+                # Get direction vector from current ee position to goal ee position for EE orientation
+                dir_rpy_to_intr = get_RPY_from_vector(goal_ee_pos - curr_ee_pos)
+
+                # Go to intermediate point
+                self.spot.move_gripper_to_point(
+                    intr_ee_pos,
+                    dir_rpy_to_intr,
+                    self.config.ARM_TRAJECTORY_TIME_IN_SECONDS,
+                    timeout_sec=10,
+                )
+
+                # Direct the gripper to face downwards
+                dir_rpy_to_goal = [0.0, np.pi / 2, 0.0]
+
+                # Go to goal point
+                self.spot.move_gripper_to_point(
+                    goal_ee_pos,
+                    dir_rpy_to_goal,
+                    self.config.ARM_TRAJECTORY_TIME_IN_SECONDS,
+                    timeout_sec=10,
+                )
+
+            # Record the success
+            local_place_target_spot = self.place_env.get_base_frame_place_target_spot()
+            local_ee_pose_spot = (
+                self.place_env.get_gripper_position_in_base_frame_spot()
+            )
+            success_list.append(
+                {
+                    "time_taken": time.time() - start_time,
+                    "success": is_position_within_bounds(
+                        local_place_target_spot,
+                        local_ee_pose_spot,
+                        self.config.SUCC_XY_DIST,
+                        self.config.SUCC_Z_DIST,
+                        convention="spot",
+                    ),
+                    "place_target": local_place_target_spot,
+                    "ee_pos": local_ee_pose_spot,
+                }
+            )
```

Comment:
@akshararai we are now using the same distance compute method for all place related environments now.

Difference being, here we are computing the distance between 2 points **specified using spot convention** while at rest all places, we are computing the distance between 2 points **specified using habitat convention** as they deal with .step() function calls .. this require all points to be as per habitat nomenclature.